testsuite/features/support/http_client.rb

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+require 'faraday'
+
+# Wrapper class for HTTP client library (Faraday)
+class HttpClient
+  def initialize(host)
+    puts 'Activating HTTP API'
+    @http_client = Faraday.new('https://' + host)
+  end
+
+  def prepare_call(name, params)
+    short_name = name.split('.')[-1]
+    call_type =
+      if short_name.start_with?('list', 'get', 'is', 'find') || ['logout', 'errata.applicableToChannels'].include?(name)
+        'GET'
+      else
+        'POST'
+      end
+    url = '/rhn/manager/api/' + name.tr('.', '/')
+    if call_type == 'GET'
+      url += '?'
+      unless params.nil?
+        params.each do |key, value|
+          url += '&' unless url[-1] == '?'
+          url += "#{key}=#{value}"
+        end
+      end
+    end
+    [call_type, url]
+  end
+
+  def call(name, params)
+    name.sub!('auth.', '') if ['auth.login', 'auth.logout'].include?(name)
+
+    # Get session cookie from previous calls
+    if params.nil?
+      session_cookie = nil
+    else
+      session_cookie = params[:sessionKey]
+      params.delete(:sessionKey)
+    end
+
+    # Call API
+    call_type, url = prepare_call(name, params)
+    answer =
+      if call_type == 'GET'
+        @http_client.get(url) do |request|
+          request.headers['Content-Type'] = 'application/json'
+          request.headers['Cookie'] = session_cookie unless session_cookie.nil?
+        end
+      else
+        @http_client.post(url) do |request|
+          request.headers['Content-Type'] = 'application/json'
+          request.headers['Cookie'] = session_cookie unless session_cookie.nil?
+          request.body = params.to_json unless params.nil?
+        end
+      end
+    raise "Unexpected HTTP status code #{answer.status}" unless answer.status == 200
+
+    # Return either new session cookie or HTTP body
+    if name == 'login'
+      session_cookie = ''
+      cookies = answer.headers['Set-cookie']
+      cookies.split(',').each do |cookie|
+        if cookie.include?('pxt-session-cookie=') && !cookie.include?('Max-Age=0;')
```

Comment:
I think it will be good to clarify better this part `&& !cookie.include?('Max-Age=0;')` with a comment, as doc for people coming to these lines later.

Reply:
ok